realtime/utils.js

```diff
@@ -408,16 +408,29 @@ function attachAspectSubject(sample, subjectModel, aspectModel) {
     sample.subject ? sample.subject : subjectModel.findOne(subOpts),
   ])
   .then((response) => {
-    let asp = response[0];
-    let sub = response[1];
-    asp = asp.get ? asp.get() : asp;
+    const ASPECT_INDEX = 0;
+    const SUBJECT_INDEX = 1;
+    let asp = response[ASPECT_INDEX];
+    let sub = response[SUBJECT_INDEX];
+
+    if (!sub) {
+      /*
+       throw error when trying to attach the subject to a sample and
+       no subject is found for the absolute path in the sample name.
+       */
+      const message = `Subject not found by Sample abs path ${subAbsPath}`;
+      throw new Error(message);
```

Comment:
So what happens now when processing a batch of samples and we can't find the subject for one of the samples in the batch?

Reply:
Well, it will just log the error and move on. The bulk upsert will not be affected because the publisher has failed.

In the story description we have the count success or failure, but as we chat yesterday, the publishing process shouldn't affect the bulk.